### PR TITLE
Add option to hide compilation buffer if all tests pass

### PR DESCRIPTION
**Pull request by mallt**
This PR adds an option that will hide the compilation buffer if all tests pass and show the test result summary and time passed in the echo area:
phpunit-hide

Thanks!

### CODE REVIEW

**Comment by zonuexe** [comment]
@mallt 
Nice work!

I have some points to review, so please wait a while.
And please forgive me for conflicting with #48 🙇

**Comment by mallt** [comment]
Thanks @zonuexe! 
No problem, I will merge the changes of #48 then after it is finished and resubmit this PR.

**Comment by zonuexe** [comment]
Maybe, follows test will be success always.

```php
<?php

final class OKTest extends \PHPUnit_Framework_TestCase
{
    public function test()
    {
        $this->assertSame(1, 2, "OK");
    }
}
```

```
-*- mode: compilation; default-directory: "~/repo/php/TetoSQL/" -*-
Compilation started at Fri Mar 10 18:54:46

stty cols 182; ~/repo/php/TetoSQL/vendor/bin/phpunit tests/OKTest.php
PHPUnit 4.8.35 by Sebastian Bergmann and contributors.
Warning:	No whitelist configured for code coverage

F

Time: 1.14 seconds, Memory: 6.00MB

There was 1 failure:

1) QueryTest::test
OK
Failed asserting that 2 is identical to 1.

/Users/megurine/repo/php/TetoSQL/tests/OKTest.php:7

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.

Compilation exited abnormally with code 1 at Fri Mar 10 18:54:48
```

**Comment by mallt** [comment]
Yes good point, I will try to make the check a bit more intelligent, thanks!

File: phpunit.el [hunk 1]
```diff
@@ -239,6 +247,23 @@ https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.anno
...
     (compile (phpunit-get-compile-command args))))
 
+(defun phpunit--hide-compilation-buffer-if-all-tests-pass (buffer status)
+  "Hide the compilation BUFFER if all tests pass.
+The STATUS describes how the compilation process finished."
+  (with-current-buffer buffer
+    (let* ((buffer-string (buffer-substring-no-properties
+                           (point-min) (point-max)))
+           (buffer-lines (s-lines buffer-string)))
+      (when-let ((ok-msg (car (cl-remove-if-not
```
Comment (zonuexe) [inline phpunit.el]: `if-let` and `when-let` are added in Emacs 25. Perhaps we should continue to support Emacs 24.3 a little more.

**Comment by zonuexe** [comment]
@mallt 
Sorry to make you wait.  Please `git rebase origin/develop`.

**Comment by mallt** [comment]
@zonuexe Thanks a lot for the comments, and sorry for the delay! I hope to update this PR soon.

**Comment by nlamirault** [comment]
Sorry for delay.

**Comment by mallt** [comment]
Thanks!

**Comment by zonuexe** [comment]
@mallt Sorry to miss it, thank you!

**Pull request merged**